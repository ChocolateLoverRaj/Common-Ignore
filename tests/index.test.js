/* global test, expect */

import generate from '../index.js'

import { dirname } from 'dirname-filename-esm'
import { readFile } from 'fs/promises'
import { join } from 'path'

const __dirname = dirname(import.meta)

test('simple setup', async () => {
  await generate({
    inputDir: './input/',
    outputDir: './output/',
    files: {
      'git.gitignore': {
        extends: [],
        output: 'test.gitignore'
      },
      'npm.npmignore': {
        extends: ['git.gitignore'],
        output: 'test.npmignore'
      }
    }
  }, __dirname)

  const checkFile = async file => {
    const contents = await readFile(file, 'utf-8')
    expect(contents).toMatchSnapshot()
  }

  await Promise.all([
    checkFile(join(__dirname, './output/test.gitignore')),
    checkFile(join(__dirname, './output/test.npmignore'))
  ])
})
