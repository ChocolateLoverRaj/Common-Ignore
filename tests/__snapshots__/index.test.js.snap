// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple setup 1`] = `
"# Dependency Tree
Line 7: git.gitignore


# Input Files

# Input: 'git.gitignore'
# Node Modules
node_modules/
"
`;

exports[`simple setup 2`] = `
"# Dependency Tree
Line 08: npm.npmignore
Line 12:             | --> git.gitignore


# Input Files

# Input: 'npm.npmignore'
# Examples
examples/

# Input: 'git.gitignore'
# Node Modules
node_modules/
"
`;
